source/fundamentals/typescript.txt

```diff
@@ -104,22 +108,30 @@ genus and color of dogs and cats:
      classification: { genus: "Canis" | "Felis"; color: string };
    }
 
-The following code snippet correctly raises a type error when specifying
-the genus of an unsupported animal in a query:
+The driver does not raise a type error for the following code sample,
+even though the value of ``classification.color`` is a boolean
+instead of a string:
 
 .. code-block:: typescript
 
-   database
-     .collection<ClassificationPet>("<your collection>")
-     .find({ "classification.genus": "Sylvilagus" });
+   await collection.findOneAndDelete({ "classification.color": false })
+
+You can enable type-checking by constructing filters as ``StrictFilter`` types.
 
-The type error raised by the preceding code snippet is as follows:
+.. warning::
+   
+   Use of the ``StrictFilter`` type to create filters is experimental and currently has
+   unknown performance implications.
 
-.. code-block:: none
+In the following code sample, the filter is assigned a
+``StrictFilter`` type. Given this filter type, the {+driver-short+}
+reports a type error because the value of ``classification.color`` is a
+boolean instead of a string:
+
+.. code-block:: typescript
 
-   No overload matches this call.
-   ...
-   Type '"Sylvilagus"' is not assignable to type 'Condition<"Canis" | "Felis">'.
+   const filterPredicate: StrictFilter<ClassificationPet> = { "classification.color": false };
+   await collection.findOneAndDelete(filterPredicate);
```

Comment:
We also had dot-notation support for UpdateFilter now moved to StrictUpdateFilter, I'm not sure if we need code snippets. It would be the same kind of migration for those that wanted to get back the strict behavior (which is also experimental):

```ts
// v4.x, compiler error
pets.updateOne({}, { $set: { "classification.color": false } })
// v5.x, no error
pets.updateOne({}, { $set: { "classification.color": false } })
// v5.x, compiler error
const update: StrictUpdateFilter<ClassificationPet> = { $set: { "classification.color": false } }
pets.updateOne({}, update)
```